packages/@o3r/core/src/utils/deep-fill/deep-fill.ts

```diff
@@ -1,35 +1,49 @@
-import {utils} from '@ama-sdk/core';
+/** Reviver Mapper to detect and create object on deepFill */
+export interface PrimitiveReviverMapper<T = any> {
+  /** Condition to fill to be determine as  */
+  condition: (data: any) => boolean;
 
-/**
- * Check if an object is a date
- * @param obj
- */
-export function isDate(obj: any) {
-  return obj instanceof utils.DateTime || obj instanceof utils.Date;
+  /**
+   * Construct the primitive type if needed
+   * @param data to be constructed
+   * @default {@see defaultConstruct}
+   */
+  construct?: (data: any) => T;
 }
 
+const defaultConstruct = (data: any) => data;
+
+const isDate = (data: any) => data instanceof Date && !isNaN(data as any);
+
 /**
  * Check if an object is not an array or a date
  * @param obj
+ * @param additionalMappers
  */
-export function isObject(obj: any) {
-  return obj instanceof Object && !Array.isArray(obj) && !isDate(obj);
+export function isObject(obj: any, additionalMappers?: PrimitiveReviverMapper[]) {
+  return obj instanceof Object && !Array.isArray(obj) && !additionalMappers?.some(({ condition }) => condition(obj)) && !isDate(obj);
```

Comment:
since condition is a method of an object, I would call it with the right context
```suggestion
  return obj instanceof Object && !Array.isArray(obj) && !additionalMappers?.some((mapper) => mapper.condition(obj)) && !isDate(obj);
```
in case the method uses `this` for any reason